.github/workflows/build.yml

```
@@ -15,21 +15,27 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [macOS-latest, ubuntu-latest]
-        python-version: [3.7, 3.8, 3.9]
+        os: [ubuntu-latest]
+        python-version: ["3.10", "3.11", "3.12"]
 
-    name: Python ${{ matrix.python-version }}, ${{ matrix.os }} build
+    name: Python ${{ matrix.python-version }} (${{ matrix.os }})
     steps:
-      - uses: actions/checkout@v2
-      - uses: conda-incubator/setup-miniconda@v2
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Setup conda
+        uses: conda-incubator/setup-miniconda@v3
         with:
-          auto-activate-base: true
-          activate-environment: true
+          miniforge-version: latest
+          auto-activate-base: false
+          activate-environment: lume-epics
+          environment-file: environment.yml
+          channels: conda-forge
+          channel-priority: strict
           python-version: ${{ matrix.python-version }}
```

You probably don't need to set up a conda env here. We can build without having the package and its dependencies installed. In fact, your next commands aren't being run in the conda shell, but instead a new venv is being created where the build dependencies get installed and the job is done, as you can see in the workflow runs [here](https://github.com/craftablescience/lume-epics/actions/runs/15175861807/job/42675909671#step:5:8)

So you can maybe simplify by only setting up python, after checking out the repo:
```suggestion
      - name: Set up Python ${{ matrix.python-version }}
        uses: actions/setup-python@v5
        with:
          python-version: ${{ matrix.python-version }}
```